scality_manila_utils/cli.py

```diff
@@ -0,0 +1,190 @@
+            break
+
+    # Drop privileges
+    if user_info and group_info:
+        log.debug('Dropping privileges to %s:%s', user_info.pw_name,
+                  group_info.gr_name)
+        os.setegid(group_info.gr_gid)
+        os.seteuid(user_info.pw_uid)
+    else:
+        msg = 'Unable to find an unprivileged user/group'
+        log.error(msg)
+        raise RuntimeError(msg)
+
+
+def main(args=None):
+    setup_logger()
+
+    if os.geteuid() != 0:
+        log.error('Invoked without superuser privileges')
+        raise RuntimeError("This program requires superuser privileges")
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
+    pre_parser.add_argument(
+        '--exports',
+        help='Path to exports file',
+        default='/etc/exports.conf'
+    )
+    pre_parser.add_argument(
+        '--root-export',
+        help='NFS export path to the ring NFS root volume',
+        default='127.0.0.1:/'
+    )
+    pre_parser.add_argument(
+        '--debug',
+        help='Set debug log level',
+        action='store_true',
+        default=False
+    )
+
+    command_parser = argparse.ArgumentParser(
+        parents=[pre_parser],
+        description='Manila exports management'
+    )
+    subparsers = command_parser.add_subparsers(help='sub-command help')
+
+    parser_create = subparsers.add_parser(
+        'create',
+        help='Prepare an export without any access grants'
+    )
+    parser_create.add_argument(
+        'export_name',
+        help='Export point to create'
+    )
+
+    parser_grant = subparsers.add_parser(
+        'grant',
+        help='Grant access to an existing filesystem, and reexport it'
+    )
+    parser_grant.add_argument(
+        'export_name',
+        help='Filesystem to grant access to'
+    )
+    parser_grant.add_argument(
+        'host',
+        help='IP address or network to grant access for'
+    )
+    parser_grant.add_argument(
+        'options',
+        help='Export options',
+        nargs='*'
+    )
+
+    parser_revoke = subparsers.add_parser(
+        'revoke',
+        help='Revoke access from an existing filesystem, and reexport it'
+    )
+    parser_revoke.add_argument(
+        'export_name',
+        help='Filesystem to revoke access from'
+    )
+    parser_revoke.add_argument(
+        'host',
+        help='IP address or network to revoke access for'
+    )
+
+    parser_check = subparsers.add_parser(
+        'check',
+        help='Check for required binaries and running services'
+    )
+
+    parser_create.set_defaults(func=Helper.add_export)
+    parser_grant.set_defaults(func=Helper.grant_access)
+    parser_revoke.set_defaults(func=Helper.revoke_access)
+    parser_check.set_defaults(func=Helper.verify_environment)
+
+    parsed_args = command_parser.parse_args(args)
+
+    # Set debug level if requested
+    if parsed_args.debug:
+        logging.root.setLevel(logging.DEBUG)
+
+    # Drop any elevated permissions
+    drop_privileges()
```

Comment:
I considered that as well, but logging must be setup before that call. Otherwise logs pertaining to `drop_privileges` will be lost. It is a small tradeoff.